packages/core/src/Table/TableCellHead.js

```diff
@@ -1,69 +1,170 @@
+                colSpan={colSpan}
+                rowSpan={rowSpan}
+                className={cx(className, {
+                    fixed,
+                    large,
+                    fixedLeft,
+                })}
+                data-test={dataTest}
+                ref={ref}
+                scope={scope}
+            >
+                <span className="container">
+                    <span className="top">
+                        <span className="label">{children}</span>
+                        {sortDirection && (
+                            <Sorter
+                                name={name}
+                                sortDirection={sortDirection}
+                                onSortIconClick={onSortIconClick}
+                            />
+                        )}
+                        {filter && (
+                            <FilterHandle
+                                name={name}
+                                active={showFilter}
+                                onFilterIconClick={onFilterIconClick}
+                            />
+                        )}
+                    </span>
+                    {filter && showFilter && (
+                        <span className="bottom">{filter}</span>
+                    )}
+                </span>
+                <style jsx>{styles}</style>
+                <style jsx>{`
+                    th {
+                        left: ${left};
+                        top: ${top};
+                        width: ${width};
+                        text-align: ${align};
+                    }
+                    .label {
+                        justify-content: ${flexboxAlignLookup[align]};
+                    }
+                `}</style>
+            </th>
+        )
+    }
 )
 
+TableCellHead.displayName = 'TableCellHead'
+
 TableCellHead.defaultProps = {
+    align: 'left',
     dataTest: 'dhis2-uicore-tablecellhead',
+    left: AUTO,
+    top: AUTO,
+    width: AUTO,
 }
 
 /**
  * @typedef {Object} PropTypes
  * @static
+ * @prop {left|center|right} [align=left]
+ * @prop {node} [children]
+ * @prop {string} [className]
  * @prop {string} [colSpan]
+ * @prop {string} [dataTest=dhis2-uicore-tablecellhead]
+ * @prop {node} [filter] The filter element (JSX)
+ * @prop {boolean} [fixed]
+ * @prop {boolean} [large]
+ * @prop {string} [left] Left or top required when fixed
+ * @prop {string} [name] Can be used to match a column with a property name
  * @prop {string} [rowSpan]
- * @prop {bool} [dense]
- * @prop {Node} [children]
- * @prop {string} [className]
- * @prop {string} [dataTest]
+ * @prop {string} [scope]
+ * @prop {boolean} [showFilter]
+ * @prop {'asc'|'desc'|null} [sortDirection]
+ * @prop {string} [top] Left or top required when fixed
+ * @prop {string} [width]
+ * @prop {function} [onFilterIconClick]
+ * @prop {function} [onSortIconClick] Sort icon click callback with `nextSortDirection` and `name` in payload
  */
 TableCellHead.propTypes = {
```

Comment:
I decided to keep this fully props driven because this ultimately is offers much more flexibility than having an internal state. But it does mean quite a few props:
- We need to be able to show the current sort direction `sortDirection` and attach a click handler to the sort-icon `onSortIconClick`
- For the filter we need to a) specify what the filter is, b) specify if we should show that filter right now, and c) attach a callback to the click on the filter-icon

In total that's 5 props to control 2 functionalities, which seems quite a lot, but I do think the gains in flexibility make it worth it. It's quite likely that at some point in time an app will need to control which filters are showing programatically.